src/blocking/spi.rs

```diff
@@ -29,26 +57,62 @@ pub trait WriteIter<W> {
         WI: IntoIterator<Item = W>;
 }
 
-/// Blocking transfer
-pub mod transfer {
+/// Blocking simultaneous read+write with separate buffers
+pub mod read_write {
+    /// Default implementation of `blocking::spi::ReadWrite<W>` for implementers of
+    /// `nonblocking::spi::FullDuplex<W>`
+    ///
+    /// If `read` is longer than `write`, `W::default()` (which is typically 0) is sent on MOSI
+    /// to fill the remaining bytes.
+    pub trait Default<W>: crate::nb::spi::FullDuplex<W> {}
+
+    impl<W, S> crate::blocking::spi::ReadWrite<W> for S
+    where
+        S: Default<W>,
+        W: Clone + core::default::Default,
+    {
+        type Error = S::Error;
+
+        fn read_write(&mut self, read: &mut [W], write: &[W]) -> Result<(), S::Error> {
```

Comment:
the reason for `transfer` is _i believe_ because the operation is simultaneous (duplex), rather than sequential (simplex). as an example in the context of the hal, `i2c::write_read` means a write _then_ a read, i believe renaming this for SPI actually makes the operation less clear.

Reply:
True. IMO the "transfer" name is a bit confusing since arguably all the SPI traits do different kinds of transfers: read transfer, write transfer, readwrite transfer... Having one of them named "transfer" feels strange.

No strong opinion though, we can name them `Transfer` and `TransferInplace` too. The main change I wanted to propose was adding the "separate buffers" transfer trait and the Read trait.

Reply:
> No strong opinion though, we can name them Transfer and TransferInplace too. The main change I wanted to propose was adding the "separate buffers" transfer trait and the Read trait.

imo at the hardware there is really only transfer, everything else is convenience above that. regardless it's different to other APIs so i'd prefer this / to keep the names different.

Reply:
The other verb I've seen used for this sort of method name is `exchange` (for example, that's what ChibiOS uses for this).

FreeRTOS uses `transfer`, STM32CubeMX uses `TransmitReceive`, libopencm3 uses `xfer`, Contiki uses `transfer`, Zephyr calls it `spi_transceive`.

Reply:
Good point with the comparison to I2C's method name @ryankurte.
I would also prefer `transfer`/`transfer_inplace` or `exchange`, `exchange_inplace`. My gut feeling is that `exchange` sounds more of a two-way transaction than `transfer`, which sounds more one-way-y, like a (monetary) bank transfer.
But anyway I am no native speaker and there seems to be a bit more usage of the `transfer` verb (thanks for the research @adamgreig!). I would be fine with either.

Reply:
Renamed to `Transfer`, `TransferInplace`. (I think "transfer" and "exchange" are equally good, but Transfer makes more sense, to minimize the name changes in 1.0)